android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -167,7 +174,24 @@ constructor(
     return if (output.hasActivity()) output else null
   }
 
+  // TODO refactor this code to remove hardcoded appended "PlanDefinition/" on https://github.com/opensrp/fhircore/issues/3386
+  private fun CarePlan.cleanPlanDefinitionCanonical() {
+    val canonicalValue = this.instantiatesCanonical.first().value
+    if (canonicalValue.contains('/').not()) {
+      this.instantiatesCanonical = listOf(CanonicalType("PlanDefinition/$canonicalValue"))
+    }
+  }
+
+  @VisibleForTesting
+  fun invokeCleanPlanDefinitionCanonical(carePlan: CarePlan) =
+    carePlan.cleanPlanDefinitionCanonical()
+
   /** Implements OpenSRP's $lite version of CarePlan & Tasks generation via StructureMap(s) */
+  @Deprecated(
+    message = "This function is deprecated, use workflowCarePlanGenerator.applyPlanDefinitionOnPatient() instead",
+    replaceWith = ReplaceWith("workflowCarePlanGenerator.applyPlanDefinitionOnPatient()"),
+    level = DeprecationLevel.WARNING
+  )
```

Comment:
@f-odhiambo did we agree on deprecating this SM approach for CarePlan generation?

Reply:
@dubdabasoduba See more here https://onaio.slack.com/archives/C01JEDB8WTU/p1721156393259029